src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -136,237 +135,188 @@ mode with existing flows and hold flows of different table-mode types"() {
 
-        and: "Both flows allow traffic"
-        def examFlow2 = new FlowTrafficExamBuilder(topology, traffExam)
-                .buildBidirectionalExam(flowHelperV2.toV1(flow2), 1000, 8)
-        withPool {
-            [examFlow1.forward, examFlow1.reverse, examFlow2.forward, examFlow2.reverse].eachParallel { direction ->
-                def resources = traffExam.startExam(direction)
-                direction.setResources(resources)
-                assert traffExam.waitExam(direction).hasTraffic()
-            }
+        and: "Involved switches pass switch validation"
+        "validate switches"(involvedSwitches)
+
+        and: "Both flows are pingable and allow traffic"
+        // This hangs (probably, blocks each other) if run in parallel mode
+        flows.each {
+            assert "is flow pingable?"(it), FLOW_ISNT_PINGABLE_ASSERTION_MESSAGE
+            assert "does flow pass traffic?"(it), FLOW_DOESNT_PASS_TRAFFIC_ASSERTION_MESSAGE
         }
 
         when: "Delete flows"
-        def flow1InfoFromDb = database.getFlow(flow.flowId)
-        def flow2InfoFromDb = database.getFlow(flow2.flowId)
-        def flowsCookies = [flow1InfoFromDb.forwardPath.cookie.value, flow1InfoFromDb.reversePath.cookie.value,
-                            flow2InfoFromDb.forwardPath.cookie.value, flow2InfoFromDb.reversePath.cookie.value]
-        [involvedSwitches[0], involvedSwitches[3]].each { sw ->
-            northbound.getSwitchRules(sw.dpId).flowEntries.each { rule ->
-                Cookie.isIngressRulePassThrough(rule.cookie) && flowsCookies << rule.cookie
-            }
-        }
-        [flow, flow2].each { flowHelperV2.deleteFlow(it.flowId) }
+        def flowsCookies = getFlowCookies(swPair, flow).addAll(getFlowCookies(swPair, flow2))
+        flows.each { flowHelperV2.deleteFlow(it.flowId) }
         def flowsAreDeleted = true
 
         then: "Flow rules are deleted from switches"
-        involvedSwitches.each { sw ->
+        switches.each { sw ->
             with(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
                 rules.findAll { it.cookie in flowsCookies }.empty
             }
         }
 
         cleanup: "Revert system to original state"
-        !flowsAreDeleted && [flow, flow2].each { it && flowHelperV2.deleteFlow(it.flowId) }
-        initSwProps && revertSwitchesToInitState(involvedSwitches, initSwProps)
+        !flowsAreDeleted && flows.each { it && flowHelperV2.deleteFlow(it.flowId) }
+        initSwProps && revertSwitchesToInitState(switches, initSwProps)
         northbound.deleteLinkProps(northbound.getLinkProps(topology.isls))
     }
 
     @Tidy
     @Tags([LOW_PRIORITY])
-    def "Single-switch flow rules are (re)installed according to switch property while rerouting,syncing,updating"() {
+    def "Single-switch flow rules are updated single- to multi-table mode when #eventName"() {
```

Comment:
so previous test also tested case when we changed table mode from multi table mode to single. But new test tests only single to multi. We lost test coverage